File: sic-app/src/app/core/services/task.service.ts
// src/app/feature/pm/services/task.service.ts

import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import type { TaskRequest, TaskResponse } from '../model/phase.model';

@Injectable({ providedIn: 'root' })
export class TaskService {
  private baseUrl = `${environment.apiBaseUrl}/api/pm/tasks`;

  constructor(private http: HttpClient) {}

  getTasksByWorkPackageId(wpId: string): Observable<TaskResponse[]> {
    return this.http.get<TaskResponse[]>(`${this.baseUrl}/work-package/${wpId}`);
  }

  getTaskById(id: string): Observable<TaskResponse> {
    return this.http.get<TaskResponse>(`${this.baseUrl}/${id}`);
  }

  createTask(data: TaskRequest): Observable<TaskResponse> {
    return this.http.post<TaskResponse>(this.baseUrl, data);
  }

  updateTask(taskId: string, data: TaskRequest): Observable<TaskResponse> {
    return this.http.put<TaskResponse>(`${this.baseUrl}/${taskId}`, data);
  }

  deleteTask(taskId: string): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${taskId}`);
  }
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/service/TaskService.java
package com.softinter.sicapi.service;


import java.util.List;
import java.util.UUID;

import com.softinter.sicapi.dto.request.TaskRequest;
import com.softinter.sicapi.dto.response.TaskResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface TaskService {
    List<TaskResponse> getTasksByWorkPackageId(UUID wpId);
    TaskResponse getTaskById(UUID taskId);
    TaskResponse createTask(TaskRequest request);
    TaskResponse updateTask(UUID taskId, TaskRequest request);
    void deleteTask(UUID taskId);
    List<TaskResponse> getAllTasksByProjectId(UUID projectId);
    List<TaskResponse> getAllTasksByBusinessId(UUID businessId, String keyword);
    Page<TaskResponse> search(UUID projectId, String keyword, Pageable pageable);
}